lib/CMakeLists.txt

```diff
@@ -31,19 +34,19 @@ if(WITH_XKB)
 endif()
 
 add_subdirectory(xpp)
-list(APPEND libs ${XPP_LIBRARIES})
-list(APPEND dirs ${XPP_INCLUDE_DIRS})
+add_library(xpp-interface INTERFACE)
+target_link_libraries(xpp-interface INTERFACE ${XPP_LIBRARIES})
+target_include_directories(xpp-interface INTERFACE ${XPP_INCLUDE_DIRS})
+list(APPEND libs xpp-interface)
 
 # }}}
 # Library: i3ipcpp {{{
 
 if(ENABLE_I3)
   add_subdirectory(i3ipcpp)
-  list(APPEND libs ${I3IPCPP_LIBRARIES})
-  list(APPEND dirs ${I3IPCPP_INCLUDE_DIRS})
+  list(APPEND libs i3ipc++)
```

Comment:
This should remain `${I3IPCPP_LIBRARIES}` for modularity.

Reply:
I can change it if you'd like, but I actually think this will be more modular.

This relates to the comment about `dirs`. `i3ipc++` is a CMake target defined [here](https://github.com/jaagr/i3ipcpp/blob/d4e4786be35b48d72dc7e59cf85ec34a90d129b5/CMakeLists.txt#L10) that actually already includes the contents of `JSONCPP_INCLUDEDIR` (defined [here](https://github.com/jaagr/i3ipcpp/blob/d4e4786be35b48d72dc7e59cf85ec34a90d129b5/CMakeLists.txt#L47)) as a property [here](https://github.com/jaagr/i3ipcpp/blob/d4e4786be35b48d72dc7e59cf85ec34a90d129b5/CMakeLists.txt#L18) and [here](https://github.com/jaagr/i3ipcpp/blob/d4e4786be35b48d72dc7e59cf85ec34a90d129b5/CMakeLists.txt#L39).

I'd also guess that the `i3ipc++` target is kept more up-to-date than the `I3IPCPP_LIBRARIES`/`I3IPCPP_INCLUDE_DIRS` variables, since if they add a new library or include directory to their project and forget to update the target, the build will fail, but that won't be true for those variables.

Reply:
```cmake
list(APPEND libs ${I3IPCPP_LIBRARIES})
```
and
```cmake
list(APPEND libs i3ipc++)
```
are completely equivalent.

`${I3IPCPP_LIBRARIES}` just contains `i3ipc++` anyway, see [here](https://github.com/jaagr/i3ipcpp/blob/d4e4786be35b48d72dc7e59cf85ec34a90d129b5/CMakeLists.txt#L46) so using the variable will keep us from duplicating the name (even if it probably never changes).